packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,632 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+
+/// Signature for a function that creates a [Widget] to be used within an
+/// [OpenContainer].
+///
+/// The `action` callback provided to [OpenContainer.closedBuilder] can be used
+/// to open the container. The `action` callback provided to
+/// [OpenContainer.closedBuilder] can be used to close the container again.
+typedef OpenContainerChildBuilder = Widget Function(
+  BuildContext context,
+  VoidCallback action,
+);
+
+/// Container, that grows to fill the screen to reveal new content when tapped.
+///
+/// While the container is closed, it shows the [Widget] returned by
+/// [closedBuilder]. When the container is tapped the container opens: It grows
+/// to fill the entire size of the surrounding [Navigator] while fading out the
+/// [Widget] returned by [closedBuilder] and fading in the [Widget] returned by
+/// [openBuilder]. When the container is closed again via the callback provided
+/// to [openBuilder] or via Android's back button the animation is reversed: The
+/// container shrinks back to its original size while the [Widget] returned by
+/// [openBuilder] is faded out and the [Widget] returned by [openBuilder] is
+/// faded back in.
+///
+// TODO(goderbauer): Add example animations and sample code.
+///
+/// See also:
+///
+///  * [Transitions with animated containers](https://material.io/design/motion/choreography.html#transformation)
+///    in the Material spec.
+class OpenContainer extends StatefulWidget {
+  /// Creates an [OpenContainer].
+  ///
+  /// All arguments except for [key] must not be null. The arguments
+  /// [closedBuilder] and [closedBuilder] are required.
+  const OpenContainer({
+    Key key,
+    this.closedColor = Colors.white,
+    this.openColor = Colors.white,
+    this.closedElevation = 1.0,
```

Comment:
If the default values of these parameters were to ever come from a theme, it's better to have them be default null. That way you can compute the actual value at build time with an expression like:
```
Color closedColor = widget.closedColor ?? OpenContainerTheme.of(context).closedColor ?? Colors.white;
```

Reply:
That should be an easy non-breaking change to make when we decide to introduce themes?

Reply:
When discussing this in person, we decided that this component would probably be themed from the outside and we can leave it as-is.